server/routes/api/urls/urls.ts

```diff
@@ -0,0 +1,165 @@
+import { differenceInMinutes, formatDistanceToNowStrict } from "date-fns";
+import { t } from "i18next";
+import Router from "koa-router";
+import { head, orderBy } from "lodash";
+import { dateLocale } from "@shared/utils/date";
+import parseDocumentSlug from "@shared/utils/parseDocumentSlug";
+import parseMentionUrl from "@shared/utils/parseMentionUrl";
+import { NotFoundError } from "@server/errors";
+import auth from "@server/middlewares/authentication";
+import { transaction } from "@server/middlewares/transaction";
+import validate from "@server/middlewares/validate";
+import { Document, User } from "@server/models";
+import { authorize } from "@server/policies";
+import { APIContext } from "@server/types";
+import { opts } from "@server/utils/i18n";
+import * as T from "./schema";
+
+const router = new Router();
+
+router.post(
+  "urls.unfurl",
+  auth(),
+  validate(T.UrlsUnfurlSchema),
+  transaction(),
+  async (ctx: APIContext<T.UrlsUnfurlReq>) => {
+    const { url, documentId } = ctx.input.body;
+    const { user } = ctx.state.auth;
+    const { transaction } = ctx.state;
+    const urlObj = new URL(url);
+    const locale = dateLocale(user.language);
+    if (urlObj.protocol === "mention:") {
```

Comment:
It would be nice to create an `UnfurlHelper` and move this logic out of the controller, they shouldn't be so large.

`UnfurlHelper.parse(url)`

Individual private methods for:

`UnfurlHelper.document(url)`

`UnfurlHelper.mention(url)`

Reply:
I don't quite get this. Is `UnfurlHelper.parse(url)` just a wrapper around protocol check plus `parseMentionUrl()/parseDocumentSlug()` or is it supposed to wrap the `findByPk` calls as well?

Reply:
This is okay for now – now that the presenter logic is separate it's small enough 👍